### ISSUE TITLE
Add generic parameter to `AjaxResponse` and `ajax`

### ISSUE THREAD

**Issue by vicrac** [ReactiveX/rxjs#5600]
## Feature Request

**Is your feature request related to a problem? Please describe.**
`ajax(params)` returns `Observable<AjaxResponse>` with field `response` typed as `any`.

**Describe the solution you'd like**
I'd like to be able to pass generic parameter to `ajax`, like:
```
ajax<{ field: string }>(params).pipe(map({ response: { field } }) => field))  // properly inferred as `Observable<string>`
```

**Describe alternatives you've considered**
Casting type each time, repetitive and annoying.

**Comment by xp44mm** [comment]
ajax have been deprecated. it be replaced by fetch.

**Comment by kwonoj** [comment]
> ajax have been deprecated. it be replaced by fetch.

It is not. core team do not have plan to deprecate ajax implementation at least in this major version and there is ongoing discussion to revisit interfaces. https://github.com/ReactiveX/rxjs/pull/5080

**Comment by cartant** [comment]
I wouldn't be in favour of this. If the `any` declaration is to be replaced, IMO it should be replaced with `unknown`. Using a type parameter to effect what can only be a type assertion is, IMO, poor practice. It should be typed as `Observable<unknown>` and callers should perform an explicit type assertion or, even better, should use a user-defined type guard.

**Comment by vicrac** [comment]
@cartant it would be the case if return type of `ajax` was the response itself. But often, we need access to additional data stored in `AjaxResponse`, like e.g. status code. If performing a type assert in this case we'd need to reflect `AjaxResponse` by ourselves, which is not very convenient. Besides, if it's not a good practice, why is there a `getJSON` function, which offers precisely this possibility (but without metadata, which is needed is my case)? IMO `AjaxResponse<T = any>` would be a perfect balance - backward compatible, and making to possible to either use type guard or passing the type kinda of "optimistically".

**Comment by cartant** [comment]
> it would be the case if return type of `ajax` was the response itself

Yeah, you are right. 🤦‍♂️ It's the `response` field that's typed as `any` and that would be better typed as `unknown`.

> ... if it's not a good practice, why is there a `getJSON` function ... ?

Practices change. With the introduction of a proper top type - `unknown` - in TS version 3, situations like this would be better typed using the top type. Using a type parameter to effect what is a type assertion is not something that I would favour. An explicit assertion would be preferable. IMO, type parameters that are unrelated to an actual parameter should be avoided.

**Comment by guiseek** [comment]
hasn't that already been done?
